feature_engine/prediction/target_mean_prediction.py

```diff
@@ -0,0 +1,212 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Union
+
+import pandas as pd
+from sklearn.base import BaseEstimator, ClassifierMixin, RegressorMixin
```

Comment:
which functionality do the ClassifierMixin and the RegressorMixin bring to the transformer?

Reply:
They both bring the `score()` method. ClassifierMixin `score()` returns the mean accuracy. RegressorMixin `score()` returns R-squared.

Reply:
but then, inheriting both would over-write each other's method.

In this case, let's not inherit the Mixins, and let's create our own score() method. If regression one thing, if classification something else. Let's take a look at the score() methods of the mixins for inspiration.